front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -142,10 +142,13 @@ const useScenarioData = (scenario: ScenarioResponse, infra: InfraWithState) => {
     [projectedTrainsById]
   );
 
-  const trainScheduleUsedForProjection = useMemo(
-    () => (trainIdUsedForProjection ? timetableItemsById.get(trainIdUsedForProjection) : undefined),
-    [trainIdUsedForProjection, timetableItems]
-  );
+  const trainScheduleUsedForProjection = useMemo(() => {
+    if (!timetableItems || timetableItems.length === 0) return undefined;
+    const found = trainIdUsedForProjection
+      ? timetableItemsById.get(trainIdUsedForProjection)
+      : undefined;
+    return found ?? timetableItems[0];
```

Comment:
I personally prefer this so we don't need to create a const, but it can be discussed.
```suggestion
if (trainIdUsedForProjection) {
    return timetableItemsById.get(trainIdUsedForProjection) ?? timetableItems[0];
  }
  return timetableItems[0];
```